### ISSUE TITLE
UI Design

### ISSUE THREAD

**Issue by JeysonFlores** [JeysonFlores/vault#1]
# Problem
The UI design has to look good and modern without loosing simplicity.

# Prior Art
This is the design Cassidy did for the [codestash](https://github.com/cassidyjames/codestash) project:
![image](https://user-images.githubusercontent.com/68255757/156259953-3bb49292-34fd-42a8-b4f4-9d0598a06fb6.png)

# Proposals
I'd suggest drop the thick headerbar and move the buttons to the bottom of the notes' list. The Search field is something I don't really  know what would be a good place for it.

**Comment by JeysonFlores** [comment]
There has been an idea in my mind since a few hours ago, what about doing a cards-style list for the notes? Something like this:
![image](https://user-images.githubusercontent.com/68255757/156288802-baab8746-b208-4d00-a550-f390d2e39a0e.png)
(of course without that side panel and stuff)

**Comment by Suzie97** [comment]
> I'd suggest drop the thick headerbar and move the buttons to the bottom of the notes' list. The Search field is something I don't really know what would be a good place for it.

I did this quick mockup.

![Artboard](https://user-images.githubusercontent.com/68198116/156615341-a1f4d5a6-ef16-4cd3-9ec1-c0a4e8488c96.png)

It's basically the same layout as CodeStash and SourceNote, but changed a little to match the newer elementary OS style, particularly the Tasks app. I think this style is tried and tested and it would be safe to go with this.

The advantages of this note according to my analysis are:
- It shows the entire content of the most recent note, when the app window is opened.
    - This would benefit a workflow where the last saved note is used the most.
- A familiar interface to Tasks, and some parts of System Settings. This is not major one imo.

Disadvantages:
- If the notes list is long, users might need to scroll through a bunch of notes to find the one they need.
- If the preview on the sidebar does not provide adequate information, users would need to click on the note to view the entire content to get the correct note.
    - This would be time consuming, when the preview is not enough for the user and they want a peek at some more content.

**Comment by Suzie97** [comment]
> There has been an idea in my mind since a few hours ago, what about doing a cards-style list for the notes?

This is an interesting idea.

Advantages:
- The card view provides more content at a glance than the sidebar list.
    - As a result, users might end up scrolling less notes to find the one they need, which is good.
- While hovering the cursor over a card, if a preview is displayed which even more content than what is shown by the card, it would improve the navigation experience.

**Comment by JeysonFlores** [comment]
> It's basically the same layout as CodeStash and SourceNote, but changed a little to match the newer elementary OS style, particularly the Tasks app

Yep, it's pretty close of what I thought originally, however one problem this generates is the fact that (in Gtk3) would make use of LibHandy's Window which is a library and a class that is not available in the C++ bindings. This would lead to 2 possible solutions:
 -  Change to another design
 -  Develop the app using Gtk4
 
 When this app is completed soon or later will be ported to Gtk4, however the fact that eOS 6 doesn't have support for it it's quite a deal-breaker, I know Gtk4 is included in the 6.1 flaptak runtime but working only using the flatpak container libraries leads to several development complications (C++ language server won't work or will treat the app as a Gtk3 app, and build times will be pretty long).

**Comment by JeysonFlores** [comment]
> While hovering the cursor over a card, if a preview is displayed which even more content than what is shown by the card, it would improve the navigation experience.

Actually I didn't think about this, I'll make a mockup to see how would it be.

**Comment by Suzie97** [comment]
> Yep, it's pretty close of what I thought originally, however one problem this generates is the fact that (in Gtk3) would make use of LibHandy's Window which is a library and a class that is not available in the C++ bindings.

Why is Handy needed? Rounded corners in the window? If that is the only case, I think adding the rounded style class to the window should work. If there are any other reasons please let me know.

**Comment by JeysonFlores** [comment]
> Why is Handy needed?

Headerbars are mandatory(and undivisible) in vanilla Gtk3, even if you let it empty and flat that side panel going from top to bottom it's not possible (as long as I know). 
Edit: If I'm not wrong, there's a trick to (kind of) achieve it. It consists to overlay a container over the headerbar. I've only seen that once in a really old elementary app, and the result was kinda buggy and ugly.

Rounded corners is not the only usage of Hdy.Window, that widget it's a clear and empty space to start building your apps without pre-defined layout designs. That's why the Hdy.Headerbar can be placed anywhere, to let the app developer put the Window's controls wherever he/she wants or needs. See: https://blogs.gnome.org/alexm/2020/04/12/on-windows-and-titlebars/

**Comment by Suzie97** [comment]
> > Why is Handy needed?
> 
> 
> 
> Headerbars are mandatory(and undivisible) in vanilla Gtk3, even if you let it empty and flat that side panel going from top to bottom it's not possible (as long as I know). 
> 
> Edit: If I'm not wrong, there's a trick to (kind of) achieve it. It consists to overlay a container over the headerbar. I've only seen that once in a really old elementary app, and the result was kinda buggy and ugly.
> 
> 
> 
> Rounded corners is not the only usage of Hdy.Window, that widget it's a clear and empty space to start building your apps without pre-defined layout designs. That's why the Hdy.Headerbar can be placed anywhere, to let the app developer put the Window's controls wherever he/she wants or needs. See: https://blogs.gnome.org/alexm/2020/04/12/on-windows-and-titlebars/

Planner had the same sidebar layout before handy was around, I had managed to implement a sidebar using planner's source. You can check out the old source code for planner and see how it's done.

**Comment by JeysonFlores** [comment]
mmm you're right, I'm having trouble compiling from source an old version of Planner but it seems Alan put a Gtk.Paned inside a a Gtk.HeaderBar. I'll open a design branch to work on the mockup

**Comment by JeysonFlores** [comment]
Replicating the code of the first release of Planner 2.0 this is what I get:
![image](https://user-images.githubusercontent.com/68255757/156907657-6d07eff7-e940-44a0-a23f-b1bd0782a6be.png)
It gives a hint of what should it be but it is not what we're looking for.

It seems Alan achieved this using hundreds of CSS code, I'm trying to modify it but I can only make it worse D:
You can try it on the `desing-mockup` branch

**Comment by JeysonFlores** [comment]
I tried other alternatives like setting the _decorated_ property of the Window to "false" but the result is even worse.
